### PR TITLE
Add a note about Higher-Ranked Trait Bounds in docs on Closures.

### PR DESCRIPTION
**Pull request by vvanders**
I hit a snag with lifetimes a few days ago and it wasn't until @birkenfeld pointed out Higher-Ranked Trait Bounds that I was able to solve the issue involving lifetimes on closure traits. This adds a small section in the book so that other users can find it.

r? @steveklabnik

### CODE REVIEW

File: src/doc/book/closures.md [hunk 1]
```diff
@@ -319,6 +319,53 @@ assert_eq!(3, answer);
...
+
+In order to say that we only need the lifetime to be valid for the invocation scope
+of the closure we can use Higher-Ranked Trait Bounds with the `for<...>` syntax:
+
+```
+fn call_with_ref<F>(some_closure:F) -> i32
+    where F: for<'a> Fn(&'a 32) -> i32 {
+```
+
+This lets the rust compiler find the minimum lifetime to invoke our closure and
```
Comment (steveklabnik) [inline src/doc/book/closures.md]: Rust, not rust

**Comment by steveklabnik** [comment]
Thanks so much for this @vvanders ! I have one small grammar nit, and also, you need to add some annotations:

```
failures:

---- Taking_closures_as_arguments_5 stdout ----
	<anon>:4:2: 4:2 error: this file contains an un-closed delimiter
<anon>:4 }
          ^
<anon>:1:11: 1:12 help: did you mean to close this delimiter?
<anon>:1 fn main() {
                   ^
<anon>:3:21: 3:23 error: expected type, found `32`
<anon>:3     where F: Fn(&'a 32) -> i32 {
                             ^~
<anon>:3:23: 3:24 error: expected one of `+`, `,`, `->`, `::`, or `{`, found `)`
<anon>:3     where F: Fn(&'a 32) -> i32 {
                               ^
error: aborting due to previous error(s)
thread 'Taking_closures_as_arguments_5' panicked at 'Box<Any>', src/librustc/session/mod.rs:167
note: Run with `RUST_BACKTRACE=1` for a backtrace.

---- Taking_closures_as_arguments_6 stdout ----
	<anon>:4:2: 4:2 error: this file contains an un-closed delimiter
<anon>:4 }
          ^
<anon>:1:11: 1:12 help: did you mean to close this delimiter?
<anon>:1 fn main() {
                   ^
<anon>:3:29: 3:31 error: expected type, found `32`
<anon>:3     where F: for<'a> Fn(&'a 32) -> i32 {
                                     ^~
<anon>:3:31: 3:32 error: expected one of `+`, `,`, `->`, `::`, or `{`, found `)`
<anon>:3     where F: for<'a> Fn(&'a 32) -> i32 {
                                       ^
error: aborting due to previous error(s)
thread 'Taking_closures_as_arguments_6' panicked at 'Box<Any>', src/librustc/session/mod.rs:167


failures:
    Taking_closures_as_arguments_5
    Taking_closures_as_arguments_6
```

The examples get run as tests, so they don't work. Adding `ignore` will fix it:

    ```ignore

Update it with that, and this should be good to go :+1:

**Comment by vvanders** [comment]
Thanks for the feedback and happy to help where I can. I'll see if I can address those and get an updated PR sent.

**Comment by vvanders** [comment]
Hey @steveklabnik, just wanted to check in here and make sure everything is good. Let me know if there's any further feedback or if you think this is okay to merge.

**Comment by steveklabnik** [comment]
@vvanders ah! Sorry about that. GitHub doesn't send a message when you push new commits; I didn't know you'd done so.

**Comment by vvanders** [comment]
No worries, glad to have it merged :).

**Pull request merged**